### ISSUE TITLE
Rubocop recommends using `each_key` on Array (function not defined on that class)

### ISSUE THREAD

**Issue by rpherbig** [bbatsov/rubocop#4856]
Rubocop is recommending that I use `each_key` on an Array, but that function is not defined on the Array class. The same issue occurs for `each_value`.

--------

## Expected behavior

No recommendation to change code.

## Actual behavior

Rubocop is recommending that I use `each_key` on an Array.

## Steps to reproduce the problem

Here is the Ruby script that causes this behavior:
```
h = { 1 => 'a' }
a = h.to_a
a.each do |k, _v|
  puts k
end
```

And the Rubocop output:
```
$ rubocop 
Inspecting 1 file
C

Offenses:

temp.rb:3:3: C: Use each_key instead of each.
a.each do |k, _v|
  ^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.0.0 i386-mingw32)
```

**Comment by Drenmi** [comment]
RuboCop does static analysis, and so we can unfortunately not know what a variable will point to at runtime. This is by far the biggest limitation of doing static analysis on a dynamic language, and there's not much we can do about these false positives, I'm afraid. (We need to favour false positives over false negatives, since the user has tools to deal with false positives.)
